ludwig/features/base_feature.py

```diff
@@ -37,6 +37,72 @@
 logger = logging.getLogger(__name__)
 
 
+class BaseFeatureMixin(ABC):
+    """Parent class for feature mixins.
+
+    Feature mixins support preprocessing functionality shared across input and output features.
+    """
+
+    @abstractstaticmethod
+    def type() -> str:
+        """Returns the type of feature this mixin supports."""
+        raise NotImplementedError
+
+    @abstractstaticmethod
+    def preprocessing_defaults() -> Dict[str, Any]:
+        """Returns dict of preprocessing defaults."""
+        raise NotImplementedError
+
+    @abstractstaticmethod
+    def preprocessing_schema() -> Dict[str, Any]:
+        """Returns schema for the preprocessing configuration."""
+        raise NotImplementedError
+
+    @abstractstaticmethod
+    def cast_column(column: DataFrame, backend) -> DataFrame:
+        """Returns a copy of the dataset column for the given feature, potentially after a type cast.
+
+        Args:
+            column: Pandas column of values.
+            backend: (Union[Backend, str]) Backend to use for feature data processing.
+        """
+        raise NotImplementedError
+
+    @abstractstaticmethod
+    def get_feature_meta(column: DataFrame, preprocessing_parameters: Dict[str, Any], backend) -> Dict[str, Any]:
+        """Returns a dictionary of feature metadata.
+
+        Args:
+            column: Pandas column of values.
+            preprocessing_parameters: Preprocessing configuration for this feature.
+            backend: (Union[Backend, str]) Backend to use for feature data processing.
+        """
+        raise NotImplementedError
+
+    @abstractstaticmethod
+    def add_feature_data(
+        feature: Dict[str, Any],
```

Comment:
maybe, as we are making some changes, we can call this feature_cfg for increased readability? In other parts of the code the name feature refers to either an input or output feature object, so maybe whenever we refer to to a dict containing the feature info we can use the _cfg suffix?